syntax = "proto3";

package envoy.extensions.tracers.lightstep.v4alpha;

import "envoy/config/core/v4alpha/base.proto";

import "udpa/annotations/status.proto";
import "udpa/annotations/versioning.proto";
import "validate/validate.proto";

option java_package = "io.envoyproxy.envoy.extensions.tracers.lightstep.v4alpha";
option java_outer_classname = "LightstepProto";
option java_multiple_files = true;
option (udpa.annotations.file_status).package_version_status = NEXT_MAJOR_VERSION_CANDIDATE;

// [#protodoc-title: LightStep tracer]

// Configuration for the LightStep tracer.
// [#extension: envoy.tracers.lightstep]
message LightstepConfig {
  option (udpa.annotations.versioning).previous_message_type =
      "envoy.config.trace.v3.LightstepConfig";

  // Available propagation modes
  enum PropagationMode {
    // Propagate trace context in the single header x-ot-span-context.
    ENVOY = 0;

    // Propagate trace context using LightStep's native format.
    LIGHTSTEP = 1;

    // Propagate trace context using the b3 format.
    B3 = 2;

    // Propagation trace context using the w3 trace-context standard.
    TRACE_CONTEXT = 3;
  }

  reserved 2;

  reserved "access_token_file";

  // The cluster manager cluster that hosts the LightStep collectors.
  string collector_cluster = 1 [(validate.rules).string = {min_len: 1}];

  // Access token to the `LightStep <https://lightstep.com/>`_ API.
  config.core.v4alpha.DataSource access_token = 4;

  // Propagation modes to use by LightStep's tracer.
  repeated PropagationMode propagation_modes = 3
      [(validate.rules).repeated = {items {enum {defined_only: true}}}];
}
